spec/consensus/proposer-based-timestamp/pbts-algorithm_002_draft.md

```diff
@@ -0,0 +1,134 @@
+So, if process `p` is the proposer of a round of height `h_p` and reads from its clock a time `now_p <= decision_p[h_p-1]`,
+it should postpone the generation of its proposal until `now_p > decision_p[h_p-1]`.
+
+Notice that monotonicity is not introduced by this proposal, being already ensured by [`bfttime`][bfttime].
+In `bfttime`, the `Timestamp` field of every `Precommit` message of height `h_p` sent by a correct process is required to be larger than `decision_p[h_p-1].time`, as one of such `Timestamp` fields becomes the time assigned to a value proposed at height `h_p`.
+
+The time monotonicity of values proposed in heights of consensus is verified by the `valid()` predicate, to which every proposed value is submitted.
+A value rejected by the `valid()` implementation is not accepted by any correct process.
+
+## Timely Proposals
+
+PBTS introduces a new requirement for a process to accept a proposal: the proposal must be `timely`.
+It is a temporal requirement, associated to a couple of synchronous assumptions regarding the behavior of processes and the network.
+
+The evaluation of the `timely` requirement requires comparing the proposal's sending time with the proposal's receiving time.
+As these two time values can be read from different clocks, at different processes, we need to assume that processes' clocks are synchronized.
+As these two times refer to two distinct events, we need to assume a minimum and a maximum real time interval between the occurrence of the two events.
+
+The two synchronous assumptions adopted to evaluate the `timely` predicate are:
+- Synchronized clocks: the values read from clocks of any two correct processes at the same instant of real time differ by at most `PRECISION`;
+- Bounded transmission delays: the real time interval between the sending of a proposal at a correct process, and the reception of the proposal at any correct process is upper bounded by `MSGDELAY`.
+
+#### **[PBTS-RECEPTION-STEP.1]**
+
+Let `now_p` be the time, read from the clock of process `p`, at which `p` receives the proposed value `v`.
+The proposal is considered `timely` by `p` when:
+1. `now_p >= v.time - PRECISION`
+1. `now_p <= v.time + MSGDELAY + PRECISION`
+
+The first condition derives from the fact that the generation and sending of `v` precedes its reception.
+The minimum receiving time `now_p` for `v` be considered `timely` by `p` is derived from the extreme scenario when
+the clock of `p` is `PRECISION` *behind* of the clock of the proposer of `v`, and the proposal's transmission delay is `0` (minimum).
+
+The second condition derives from the assumption of an upper bound for the transmission delay of a proposal.
+The maximum receiving time `now_p` for `v` be considered `timely` by `p` is derived from the extreme scenario when
+the clock of `p` is `PRECISION` *ahead* of the clock of the proposer of `v`, and the proposal's transmission delay is `MSGDELAY` (maximum).
+
```

Comment:
Shall we add a remark w.r.t. the issue of proposal message vs. block parts.

Reply:
Should we detail this there?

Reply:
Let's put it in an issue, and merge this version now...